packaging_automation/update_docker.py

```diff
@@ -0,0 +1,147 @@
+    header = f"### citus-docker v{project_version}.docker ({datetime.strftime(datetime.now(), '%B %d,%Y')}) ###\n"
+    citus_bump_str = f"\n* Bump Citus version to {project_version}\n"
+    postgres_bump_str = f"\n* Bump PostgreSQL version to {postgres_version}\n"
+
+    changelog_entry = f"{header}{citus_bump_str}"
+    if postgres_version:
+        changelog_entry = f"{changelog_entry}{postgres_bump_str}"
+    changelog_entry = f"{changelog_entry}\n"
+    return changelog_entry
+
+
+def update_changelog(project_version: str, exec_path: str, postgres_version: str = ""):
+    latest_changelog = get_new_changelog_entry(project_version, postgres_version)
+    changelog_file_path = f"{exec_path}/CHANGELOG.md"
+    with open(changelog_file_path, "r+") as reader:
+        if not (f"({project_version}" in reader.readline()):
+            reader.seek(0, 0)
+            old_changelog = reader.read()
+            changelog = f"{latest_changelog}{old_changelog}"
+            reader.seek(0, 0)
+            reader.write(changelog)
+        else:
+            raise ValueError("Already version in the changelog")
+
+
+def update_all_docker_files(project_version: str, tools_path: str, exec_path: str, postgres_version: str):
+    template_path = f"{tools_path}/packaging_automation/templates/docker"
+    pkgvars_file = f"{exec_path}/pkgvars"
+
+    if postgres_version:
+        update_pkgvars(project_version, template_path, pkgvars_file, postgres_version)
+
+    persisted_postgres_version = read_postgres_version(pkgvars_file)
+
+    update_docker_file_for_latest_postgres(project_version, template_path, exec_path, persisted_postgres_version)
+    update_regular_docker_compose_file(project_version, template_path, exec_path)
+    update_docker_file_alpine(project_version, template_path, exec_path, persisted_postgres_version)
+    update_docker_file_for_postgres12(project_version, template_path, exec_path)
+    update_changelog(project_version, exec_path, postgres_version)
+
+
+def read_postgres_version(pkgvars_file: str) -> str:
+    if os.path.exists(pkgvars_file):
+        with open(pkgvars_file, "r") as reader:
+            content = reader.read()
+            lines = content.splitlines()
+            for line in lines:
+                if line.startswith("latest_postgres_version"):
+                    line_parts = line.split("=")
+                    if len(line_parts) != 2:
+                        raise ValueError("keys and values should be seperated with '=' sign")
+                    else:
+                        postgres_version = line_parts[1]
+                else:
+                    raise ValueError("pkgvars file should include a line with key latest_postgres_version")
+    else:
+        # Setting it because pkgvars does not exist initially
+        postgres_version = "13.2"
+    return postgres_version
+
+
+def update_pkgvars(project_version: str, template_path: str, pkgvars_file: str, postgres_version: str):
+    if postgres_version:
```

Comment:
I think we can remove `if postgres_version:` check here, since the caller already checks that.

Reply:
This function is only used to update `postgres_version`, right ?
I think we should not even call this function if `postgres_version` is empty, so I think we should not need to test this function with `postgres_version=None`.

What do you think ?

Reply:
Removed the control